prysm/io.py

```diff
@@ -1363,3 +1363,96 @@ def read_sigfit_rigidbody(file):
             'dR': dR
         }
     return out
+
+def read_codev_int(filename):
+    """Read the content of a CodeV INT (.int/.cvg) file.
+    
+    Parameters
+    ----------
+    file : path_like
+        path to a file
+    
+    Returns
+    -------
+    'dict'
+        dictionary with keys: phase, intensity, meta
+
+    """
+    with open(filename, 'rt') as f:
+        # skip headerlines starting with "!"
+        while True:
+            title = f.readline()
+            if not title.lstrip().startswith('!'):
+                break
+        # read parameter line
+        param = f.readline().split()
+        # parse parameters
+        if param[0] == 'GRD':
+            num_cols = int(param[1])
+            if param[2] == 'R':
+                #radial grid data
+                #not implemented
+                raise ValueError('Radial Grid Data INT-File type not supported.')
+            else:
+                num_rows = int(param[2])
+            datatype = param[3]    #SUR | WFR | FIL
+            #need to test now because different orders are allow
+            p=4
+            while p < len(param):
+                if param[p].upper() == 'WVL':
+                    wavelength = float(param[p+1])
+                if param[p].upper() == 'NMB':
+                    pass
+                if param[p].upper() == 'SSZ':
+                    scale_size = float(param[p+1])
+                if param[p].upper() == 'NDA':
+                    nda = int(param[p+1])
+                if param[p].upper() == 'XSC':
+                    x_scale = float(param[p+1])
+                p += 1    
+            #read grid data
+            grdlist=[]
+            while True:
+                line = f.readline()
+                if not line:
+                    break
+                int_line = [int(i) for i in line.split()]
+                grdlist.extend(int_line)
+            if not len(grdlist) == num_cols*num_rows:
+                raise ValueError('Error during INT-file import. Length of list does not match row*cols')
+            if (datatype == 'WFR') or (datatype == 'SUR') :
+                # return values in nm (zygo convention, fit to prysm)
+                phase = e.array(grdlist).astype(config.precision)
```

Comment:
can you express the reading above this as one or a few numpy calls?  I can imagine genfromtxt can do the reading, ravel it, then reshape it.  If it can be done in a few numpy calls, it will be faster and less error prone.

Reply:
I will take a look at genfromtxt.